StRoot/St_base/St.h

```diff
@@ -0,0 +1,48 @@
+/***************************************************************************
```

Comment:
> Can you illustrate with an example how you propose to use the new St class in our existing macros?

Just like @plexoos I am puzzled by the use case of this class. From the commit message where it says

> In result your macro does not work if you have any problem in subbmacro, which is not even called.

it can be inferred that we have some malformed macros somewhere. Makes me wonder if we should look into fixing those instead. In any case it's really hard to review a piece of code without seeing an example of its use.